src/resources/Validator.js

```diff
@@ -1,21 +1,49 @@
 'use strict';
 const _ = require('lodash');
+const {ObjectId} = require('mongodb');
 const {
   ProcessTargets,
   process,
   interpolateErrors,
   escapeRegExCharacters,
   serverRules,
-  Utils,
+  Utils: CoreUtils,
 } = require('@formio/core');
 const {evaluateProcess} = require('@formio/vm');
-const util = require('../util/util');
+const Utils = require('../util/util');
 const fetch = require('@formio/node-fetch-http-proxy');
 const debug = {
   validator: require('debug')('formio:validator'),
   error: require('debug')('formio:error')
 };
 
+// Promisify cache load form.
+function loadFormById(cache, req, formId) {
+  return new Promise((resolve, reject) => {
+    cache.loadForm(req, null, formId, (err, resource) => {
+      if (err) {
+        return reject(err);
+      }
+      return resolve(resource);
+    });
+  });
+}
+
+// Promisify submission model find.
+function submissionQueryExists(submissionModel, query) {
+  return new Promise((resolve, reject) => {
+    submissionModel.find(query, (err, result) => {
+      if (err) {
+        return reject(err);
+      }
+      if (result.length === 0 || !result) {
+        return resolve(false);
+      }
+      return resolve(true);
+    });
+  });
+}
+
```

Comment:
You could probably leverage node's `promisify` [utility function](https://nodejs.org/docs/latest/api/util.html#utilpromisifyoriginal) here.  It converts functions with callback as last parameter to a promise.  We've done it before with cache functions.  If the method signature doesn't match, you have to wrap the function in a lambda with the args reordered.